mdp/hinet/layer.py

```diff
@@ -116,7 +134,7 @@ def _get_supported_dtypes(self):
         return list(types)
 
     def is_trainable(self):
-        return any(node.is_trainable() for node in self.nodes)
+        return self._is_trainable
```

Comment:
Well it is mostly for consistency. When creating a layer, we need to check not just whether the individual nodes are trainable but if there are any remaining training phases. For example, a layer made up of trained SFAs would still return True when is_trainable is called. But in principle the layer is a non-trainable node.